jazkarta/shop/utils.py

```diff
@@ -113,12 +121,18 @@ def send_mail(subject, message, mfrom=None, mto=None):
         realname, replyaddr = email.utils.parseaddr(mfrom)
         msg['Reply-To'] = Header(mfrom, 'utf-8')
 
-    registry = getUtility(IRegistry)
-    mail_settings = registry.forInterface(IMailSchema, prefix='plone')
+    if REGISTRY_HAS_MAIL_SETTINGS:
+        registry = getUtility(IRegistry)
+        mail_settings = registry.forInterface(IMailSchema, prefix="plone")
+        if not realname:
+            realname = mail_settings.email_from_name
+        mfrom = email.utils.formataddr((realname, mail_settings.email_from_address))
+    else:
+        portal = get_site()
+        if not realname:
+            realname = portal.getProperty('email_from_name')
+        mfrom = email.utils.formataddr((realname, portal.getProperty('email_from_address')))
```

Comment:
I would probably move this to a separate function for getting the from name and address.